app/services/browser_support.rb

```diff
@@ -38,16 +37,29 @@ def clear_cache!
 
     attr_reader :cache
 
+    def matchers_for_browser(browser)
+      if browser.ios?
+        matchers.slice(:ios_saf)
+      elsif browser.platform.android_webview?
+        matchers.slice(:android)
+      else
+        matchers
+      end
+    end
+
     def matchers
       @matchers ||= browser_support_config.flat_map do |config_entry|
         key, version = config_entry.split(' ', 2)
+        key = key.to_sym
         browser_matcher = BROWSERSLIST_TO_BROWSER_MAP[key]
         next [] if !browser_matcher
 
         low_version, _high_version = version.split('-', 2)
         low_version = nil if !numeric?(low_version)
-        proc { |browser| browser_matcher.call(browser, low_version && ">= #{low_version}") }
-      end
+        version_test = low_version && ">= #{low_version}"
+        matcher = proc { |browser| browser_matcher.call(browser, version_test) }
+        [[key, matcher]]
```

Comment:
the double array seems weird to me? are we trying to avoid a second iteration if we did `next nil` + `.compact.to_h`?

Reply:
> the double array seems weird to me? are we trying to avoid a second iteration if we did `next nil` + `.compact.to_h`?

Subconsciously I might have been trying to avoid that, yeah. I also don't know if I find it especially more or less clear one way or the other.

Another option is a reduce-y like operation?

```diff
diff --git a/app/services/browser_support.rb b/app/services/browser_support.rb
index 65983341e2..6ce3030d58 100644
--- a/app/services/browser_support.rb
+++ b/app/services/browser_support.rb
@@ -50,3 +50,3 @@ class BrowserSupport
     def matchers
-      @matchers ||= browser_support_config.flat_map do |config_entry|
+      @matchers ||= browser_support_config.each_with_object({}) do |config_entry, result|
         key, version = config_entry.split(' ', 2)
@@ -54,3 +54,3 @@ class BrowserSupport
         browser_matcher = BROWSERSLIST_TO_BROWSER_MAP[key]
-        next [] if !browser_matcher
+        next result if !browser_matcher
 
@@ -60,4 +60,5 @@ class BrowserSupport
         matcher = proc { |browser| browser_matcher.call(browser, version_test) }
-        [[key, matcher]]
-      end.to_h
+        result[key] = matcher
+        result
+      end
     end
```